xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp

```diff
@@ -1421,12 +1411,12 @@ void CActiveAE::Configure(AEAudioFormat *desiredFmt)
       !CompareFormat(m_sinkBuffers->m_inputFormat, sinkInputFormat) ||
       m_sinkBuffers->m_format.m_frames != m_sinkFormat.m_frames))
   {
-    m_discardBufferPools.push_back(m_sinkBuffers);
-    m_sinkBuffers = NULL;
+    m_discardBufferPools.push_back(std::move(m_sinkBuffers));
```

Comment:
Take care here - the code path continues and next lines checks for !m_sinkBuffers, which won't be correct anymore now when it is not NULLed.

Reply:
that is `std::move` semantics and will leave `m_sinkBuffers` as `nullptr`

see: https://godbolt.org/z/nrx7qxcf3
and: https://stackoverflow.com/a/36071266

Reply:
it seems to be guaranteed nullptr for a moved from unique_ptr but not in general.
https://wiki.sei.cmu.edu/confluence/display/cplusplus/EXP63-CPP.+Do+not+rely+on+the+value+of+a+moved-from+object

why not just add a comment?

Reply:
@lrusak thanks much, I did not know that!

Reply:
@lrusak, your commit operates flawlessly on my system.

Reply:
> it seems to be guaranteed nullptr for a moved from unique_ptr but not in general.
> 
> https://wiki.sei.cmu.edu/confluence/display/cplusplus/EXP63-CPP.+Do+not+rely+on+the+value+of+a+moved-from+object
> 
> 
> 
> why not just add a comment? 

@howie-f from the link you posted for `std::unique_ptr`:

> The moved-from object is guaranteed to refer to a null pointer value, per [unique.ptr], paragraph 4 [ISO/IEC 14882-2014].

Reply:
@lrusak yea, i thought probably add a comment to the code :)
it‘s all good like it is now